src/devices/badger_orion_endpoint.c

```diff
@@ -0,0 +1,228 @@
+
+All models above may not be compatible with this decoder as few of them are using mobile frequency, like LTE models.
+
+Frequency Hopping Spread Spectrum Intentional Radiators Operating within the 902-928MHz band, source : https://fcc.report/FCC-ID/GIF2014W-OSE/2499315
+
+2 Hopping options, Fixed Mode or Mobile Mode:
+- Fixed Mode, 400 Khz between channels, total of 50 channels from 904.56 Mhz to 924.56 Mhz
+- Mobile Mode, 400.55 Khz between channels, total of 48 channels from 904.45 Mhz to 923.675 Mhz
+
+Frequency channel changed every 150 secondes (#2995)
+
+- Message is encoded using IBM Whitening Algorithm.
+
+Flex decoder:
+
+    rtl_433 -X "n=orion_endpoint,m=FSK_PCM,s=10,l=10,r=1000,preamble=aaaaec62ec62" 2>&1 | grep codes
+
+    codes: {185}eb e1 1d 9a ed 6d 4a 4d e8 71 93 3a 78 23 57 0a ae ce 2d d8 7d 3f 4e 0
+    codes: {184}eb e1 1d 9a ed 5f 99 7c e8 71 92 31 42 62 14 0a b3 95 6e d8 7d 59 7e
+
+Data layout:
+
+    Byte Position   0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25
+    Sample         eb e1 1d 9a ed 6d 4a 4d e8 71 93 3a 78 23 57 0a ae ce 2d d8 7d 3f 4e 0
+    unwhiten       14 00 00 00 00 e8 79 69 02 0b 41 03 08 b4 00 00 fa b3 00 00 10 32 f4 4
+                   LL 11 11 11 11 SS SS SS SS xx |x xx RR RR RR RR DD DD DD DD 22 CC CC TT TT TT
+                                                 |
+                                              +--+---+
+                                              | xxLx |
+                                              +------+
+
+
+- LL: {8} Message length except CRC, mostly 0x14 = 20 bytes, to be confirmed.
+- II:{32} Fixed value, 0x00000000, could be reverse flow water counter ?
+- SS: {32} Serial Number, little-endian value
+- xx: Unknown, values look fixed and depends on the model, could be flags also, battery level too, to be guessed
+- L:{1} Leak
+- xx: other unknown values, flags, model, unit, battery low ? to be guessed.
+- RR: {32} Reading value, scale 10 gallon, little-endian value
+- DD: {32} Daily Reading Value, scale 10 gallon, little-endian value
+- FF:{8} Fixed value, always 0x10
+- CC:{16} CRC-16, poly 0x8005, init 0xFFFF, final XOR 0x0000, from previous 21 bytes.
+- TT: Trailing bits
+
+*/
+static int orion_endpoint_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble_pattern[6] = {0xaa, 0xaa, 0xec, 0x62, 0xec, 0x62};
+
+    // ibm_whiten() function added to bit_util but requires CPU cycles, as we just need XOR with fixed value this will take less resources.
+    uint8_t const ibm_whiten_key[23] = {0xff, 0xe1, 0x1d, 0x9a, 0xed, 0x85, 0x33, 0x24, 0xea, 0x7a, 0xd2, 0x39, 0x70, 0x97, 0x57, 0x0a, 0x54, 0x7d, 0x2d, 0xd8, 0x6d, 0x0d, 0xba};
+
+    uint8_t b[23];
+
+    if (bitbuffer->num_rows > 1) {
+        decoder_logf(decoder, 1, __func__, "Too many rows: %d", bitbuffer->num_rows);
+        return DECODE_FAIL_SANITY;
+    }
+    int msg_len = bitbuffer->bits_per_row[0];
+
+    if (msg_len > 290) {
```

Comment:
Optimization:  also reject message which are too short: shorter than `sizeof(preamble) * 8 + 184`. I've seen a ton of broadcasts captured which are shorter than this, so, this will help save CPU cycles on those.